pkg/pgxconn/pgx_conn.go

```diff
@@ -37,21 +47,63 @@ type connImpl struct {
 	Conn *pgxpool.Pool
 }
 
+type pgxRows struct {
+	pgx.Rows
+	sqlQuery  string
+	args      []interface{}
+	startTime time.Time
+	loggged   bool
+}
+
 func (p *connImpl) Close() {
 	conn := p.Conn
 	p.Conn = nil
 	conn.Close()
 }
 
-func (p *connImpl) Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error) {
-	return p.Conn.Exec(ctx, sql, arguments...)
+func (p *pgxRows) Next() bool {
+	if !p.loggged {
+		p.loggged = true
+		LogQueryStats(p.sqlQuery, p.startTime, p.args)()
+	}
+	return p.Rows.Next()
+}
+
+func (p *pgxRows) Scan(dest ...interface{}) error {
+	return p.Rows.Scan(dest...)
+}
+
+func (p *pgxRows) Err() error {
+	return p.Rows.Err()
+}
+
+func (p *pgxRows) Close() {
+	p.Rows.Close()
+}
+
+// calc SQL query execution time
+func LogQueryStats(sql string, startTime time.Time, args ...interface{}) func() {
+	if startTime.IsZero() {
+		startTime = time.Now()
+	}
+	return func() {
+		log.Debug("msg", fmt.Sprintf("time taken by SQL query: %s with args: %v is %v", sql, args, time.Since(startTime)))
```

Comment:
```suggestion
		log.Debug("msg", "SQL query timing", "query", sql, "args", args, "time",  time.Since(startTime))
```

Reply:
Agrs are gonna be long. Lets do

```suggestion
		log.Debug("msg", "SQL query timing", "duration", time.Since(startTime), "query", sql, "args", args)
```

Reply:
Also, @VineethReddy02 can you please paste a sample output of how it looks?

Reply:
I think the time consumed at the end will b easy to analyze the time taken quickly rather than finding it in the middle of the log lines.